### PR TITLE
directly handle empty constraints

### PR DESCRIPTION
**Pull request by MaxOstrowski**
Sorry for being stupid again.
I tried to handle empty constraints directly by not introducing them into the DLPropagator.
When writing the unit test I tried to check for choices=0.
To figure out where this statistic is hidden i tried to enumerate the keys of the statistics map.
Unfortunately, I do get a Segfault when trying to do so.
What is the intended behaviour to read out the statistics keys.

### CODE REVIEW

File: libclingo-dl/tests/solve.cc [hunk 1]
```diff
@@ -190,6 +190,33 @@ TEST_CASE("solving", "[clingo]") {
...
+            auto result = solve(theory, ctl);
+            REQUIRE(result == (ResultVec{{}}));
+            StatisticsType type= ctl.statistics().type();
+            if (type == StatisticsType::Value)
+                std::cout << "Value " << std::endl;
+            if (type == StatisticsType::Map)
+                std::cout << "Map " << std::endl;
+            if (type == StatisticsType::Array)
+                std::cout << "Array " << std::endl;
+            for (const auto i : ctl.statistics().keys()) {
```
Comment (rkaminsk) [inline libclingo-dl/tests/solve.cc:211]: The `Statistics` object in the loop gets destroyed after the keys have been returned. This invalidates the keys. The require in the following line is okay though.

Comment (MaxOstrowski) [reply libclingo-dl/tests/solve.cc:211]: Thanks.

File: libclingo-dl/clingo-dl/propagator.hh [hunk 2]
```diff
@@ -1316,7 +1316,15 @@ public:
         }
         auto u_id = map_vert(Clingo::Number(0));
         auto v_id = map_vert(Clingo::Number(0));
-        if (covec.size() == 1) {
+        if (covec.size() == 0) {
+            int eval_lit = (0 <= rhs) ? 1 : -1;
+            if (!init.add_clause({-literal, eval_lit})) return false;
```
Comment (rkaminsk) [inline libclingo-dl/clingo-dl/propagator.hh:1319]: Please use braces. Just looking at it I would turn it into:
```cpp
return init.add_clause({-literal, eval_lit}) && (!strict || init.add_clause({-eval_lit, literal}))
```
Looking a bit more I would write:
```cpp
if (rhs > 0) {
    return init.add_clause({-literal})
}
return !strict || init.add_clause({literal})
```
It is probably also possible to make the above shorter. But I think we are getting a nice balance between readability and compactness here.

I think there is a bug with `-literal` in your code, which I tried to fix.

Comment (MaxOstrowski) [reply libclingo-dl/clingo-dl/propagator.hh:1319]: I think there is something wrong with your last code-snippet (all unit tests trigger).
If rhs > 0, this means that the constraint is true.
As we have an implication `literal->T` this is tautology, and no clause should be posted (literal can either be true or false.
In the strict case we would also have `T->literal`, which means that we would need to add clause `{literal}`.

If rhs < 0 we have `literal->F`, we would need to add clause `{-literal}`.
Again, for strict we have F->literal, which results in no clauses.
Is this right ?

Comment (MaxOstrowski) [reply libclingo-dl/clingo-dl/propagator.hh:1319]: I end up with:
```
            if (rhs < 0) {
                return init.add_clause({-literal});
            }
            return !strict || init.add_clause({literal});
```

Comment (MaxOstrowski) [reply libclingo-dl/clingo-dl/propagator.hh:1319]: Sorry, i already merged before, so we are back to the normalize branch again.

**Pull request merged**